### PR TITLE
fix(config): reject malformed integer settings

### PR DESCRIPTION
**Pull request by ihor-sokoliuk**
## Summary
- reject malformed whole-string values for selected cache, URL output/fetch-timeout, HTTP-port, and search integer settings
- preserve existing defaults, bounds, blank handling, and warnings through one shared strict parser
- add regression coverage for suffix-bearing values such as `10s`, `500x`, and `3000x`

## Verification
- `npm run test:coverage` (571/571; 96.38% lines, 93.05% branches)
- `npm run build`
- `npm run lint`
- `npm run test:e2e` (11/11, including live SearXNG)
- `npm audit --audit-level=moderate` (0 vulnerabilities)
- `npm run verify:packed-consumer` (adapter 2.0.12, audit 0, tools 4)
- `git diff --check`

### CODE REVIEW

File: src/env-int.ts [hunk 1]
```diff
@@ -4,13 +4,18 @@
  * cache (`search-cache.ts`) so their env-parsing behavior stays identical.
  */
 
+export function parseStrictInteger(value: string): number | undefined {
+  const parsed = Number(value.trim());
+  return Number.isInteger(parsed) ? parsed : undefined;
+}
```
[the remark this answers is not in the document]
Comment (ihor-sokoliuk) [reply src/env-int.ts:14]: Addressed in 3c80694 — verified valid. The parser now accepts only signed base-10 digits and rejects values outside JavaScript's safe-integer range; regressions cover 5.0, 1e3, 0x10, and 9007199254740993.

**Comment by ihor-sokoliuk** [comment]
Review round 1: tightened the shared parser to signed base-10 digits within the safe-integer range. Added regressions for decimal notation, scientific notation, hexadecimal notation, and unsafe integers. No findings were left unchanged.

**Comment by ihor-sokoliuk** [comment]
Review round 2: added guaranteed cleanup to all new search regressions and updated the shared parser's module comment. The separate URL content-length parser remains unchanged here because it is outside this PR's declared files and is tracked as a dedicated follow-up. No other low-confidence findings were accepted.

**Comment by ihor-sokoliuk** [comment]
Review round 3: replaced the fixed log sleep with bounded polling, aligned the CLI timeout with the existing startup allowance, and narrowed the PR wording so it does not claim the separately scoped URL content-length parser. No findings were dismissed; the URL content-length behavior remains a dedicated follow-up.

**Comment by ihor-sokoliuk** [comment]
Review round 4: Copilot reviewed the exact head c674c02 across all 8 changed files and reported no comments or suppressed findings. Fresh verification also passed: 571 tests, build, lint, 11 live E2E checks, zero audit findings, packed-consumer adapter 2.0.12/audit 0/tools 4, and diff check.
